gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionProps.flex

```diff
@@ -67,7 +66,7 @@ GROUP_PART=[^.:=\ \n\t\f\\] | "\\ "
 
 <WAITING_VALUE> {WHITE_SPACE}+                              { yybegin(WAITING_VALUE); return TokenType.WHITE_SPACE; }
 
-<WAITING_VALUE> {FIRST_VALUE_CHARACTER}{VALUE_CHARACTER}*   { yybegin(YYINITIAL); return VersionPropsTypes.VERSION; }
+<WAITING_VALUE> {VALUE_CHARACTER}+                          { yybegin(YYINITIAL); return VersionPropsTypes.VERSION; }
```

Comment:
I think it would be nice to add some tests for the lexer & parser. I see that Intellij has some frameworks to do that: https://plugins.jetbrains.com/docs/intellij/parsing-test.html and https://github.com/JetBrains/intellij-community/blob/idea/242.22855.74/platform/testFramework/src/com/intellij/testFramework/LexerTestCase.java.

I tested now the case where you have something like: 
```
com.fasterxml.jackson.core:jackson-databind = 2.17.2\
com.google.auto.service:auto-service = 1.1.1
```

I am seeing this syntax error (but it is quite hidden): `VersionPropsTokenType.COMMENT, VersionPropsTokenType.CRLF or VersionPropsTokenType.GROUP_PART expected, got '=' `.

<img width="714" alt="Screenshot 2024-09-25 at 3 01 00 PM" src="https://github.com/user-attachments/assets/d0b3c747-364e-4642-8b6c-072da09dac67">

With the new change from:https://github.com/palantir/gradle-consistent-versions-idea-plugin/pull/10/files#diff-b8f5308d750c8b00dcf2749190e1362cc968661cc7597f358224253cee97e9d5R69 we will silently fail - we won't run writeVersionsLock and the user won't see clearly that the versions.props file is invalid.

Reply:
I'll look at adding some tests now

I originally added the check for bad character in the `wVL` change as previously `wVL` would run on every autosave which often coincided with half typed lines and cause lots of annoying errors, I wonder if the best thing would be a popup of some sort?

Reply:
So I managed to fix the lexer so that it didnt include end of line comments and wrapping with \ but failed on the tests.